components/script/script_module.rs

```diff
@@ -1434,18 +1501,26 @@ fn fetch_an_import_module_script_graph(
     global: &GlobalScope,
     module_request: RawHandle<*mut JSObject>,
     reference_private: RawHandleValue,
-    base_url: ServoUrl,
-    options: ScriptFetchOptions,
     promise: Rc<Promise>,
     can_gc: CanGc,
 ) -> Result<(), RethrowError> {
     // Step 1.
     let cx = GlobalScope::get_cx();
-    rooted!(in(*cx) let specifier = unsafe { GetModuleRequestSpecifier(*cx, module_request) });
-    let url = ModuleTree::resolve_module_specifier(*cx, &base_url, specifier.handle().into());
+    let specifier = unsafe {
+        jsstring_to_str(
+            *cx,
+            ptr::NonNull::new(GetModuleRequestSpecifier(*cx, module_request)).unwrap(),
+        )
+    };
```

Comment:
I think you still want to root the `specifier`, and then use `jsstring_to_str` on it.